src/Csv.php

```diff
@@ -889,10 +889,30 @@ public function load_data($input = null) {
 
         if (is_null($input)) {
             $file = $this->file;
-        } elseif (file_exists($input)) {
-            $file = $input;
         } else {
-            $data = $input;
+            $inputLengthCondition = mb_strlen($input) > PHP_MAXPATHLEN;
+            if ($inputLengthCondition) {
+                $baseString = mb_split(DIRECTORY_SEPARATOR,
+                    mb_substr($input, 0, PHP_MAXPATHLEN));
+
+                end($baseString);
+                unset($baseString[key($baseString)]);
+                $baseString = implode(DIRECTORY_SEPARATOR, $baseString);;
+                if (file_exists($baseString)) {
+                    Throw new FileNotFoundException(
```

Comment:
1. The keyword Throw has to be in lower case.
2. I find the idea of parsing a string longer than PHP_MAXPATHLEN as a path a bit ridiculous. Did you run into this with an actual path or is it hypothetical code?